### PR TITLE
Iterate on accessibility checker docs

### PR DESCRIPTION
**Pull request by thibaudcolas**
Follow-up to #10029 – improving different aspects of the new docs section.

### CODE REVIEW

File: docs/advanced_topics/accessibility_considerations.md [hunk 1]
```diff
@@ -128,14 +128,14 @@ A number of built-in tools and additional resources are available to help create
 
 ### Built-in accessibility checker
 
-Wagtail includes an accessibility checker built into the [user bar](wagtailuserbar_tag). The checker can help authors create more accessible websites in compliance with the Web Content Accessibility Guidelines (WCAG) 2.0 and the Authoring Tool Accessibility Guidelines (ATAG) 2.0.
+Wagtail includes an accessibility checker built into the [user bar](wagtailuserbar_tag). The checker can help authors create more accessible websites following best practices and accessibility standards like [WCAG](https://www.w3.org/WAI/standards-guidelines/wcag/).
```
Comment (thibaudcolas) [inline docs/advanced_topics/accessibility_considerations.md:131]: - Generally "compliance" is reserved for legal compliance.
- WCAG 2.0 is relevant but also 2.1, soon 2.2, and more generally best practices
- The checker helps _Wagtail_ meet ATAG requirements, not the websites created with Wagtail

File: docs/advanced_topics/accessibility_considerations.md [hunk 2]
```diff
@@ -128,14 +128,14 @@ A number of built-in tools and additional resources are available to help create
...
 
-The checker is based on the [Axe](https://github.com/dequelabs/axe-core) testing engine and scans the loaded page for errors, displaying the results in the GUI.
+The checker is based on the [Axe](https://github.com/dequelabs/axe-core) testing engine and scans the loaded page for errors.
 
-By default, the checker includes the following rules that are among the most common and critical accessibility issues to be violated by editors in Wagtail:
+By default, the checker includes the following rules that find common accessibility issues in authored content:
```
Comment (thibaudcolas) [inline docs/advanced_topics/accessibility_considerations.md:135]: I think it reads better to focus on the issues being in the content than on people violating rules.

File: docs/advanced_topics/accessibility_considerations.md [hunk 3]
```diff
@@ -128,14 +128,14 @@ A number of built-in tools and additional resources are available to help create
...
 
 -   `empty-heading`: This rule checks for headings with no text content. Empty headings are confusing to screen readers users and should be avoided.
--   `p-as-heading`: This rule checks for paragraphs that are styled as headings. Paragraphs should not be styled as headings, as this can cause confusion for users who rely on headings to navigate content.
+-   `p-as-heading`: This rule checks for paragraphs that are styled as headings. Paragraphs should not be styled as headings, as they don’t help users who rely on headings to navigate content.
```
Comment (thibaudcolas) [inline docs/advanced_topics/accessibility_considerations.md:138]: It’s not really confusing, just the headings that should be there aren’t there for those users without them having a way to tell either way.

File: docs/advanced_topics/accessibility_considerations.md (hunk 2 above)
Comment (laymonage) [inline docs/advanced_topics/accessibility_considerations.md:135]: Maybe it's just me but the "that" in this case reads a bit off to me, I find it easier with "to" instead:

```suggestion
By default, the checker includes the following rules to find common accessibility issues in authored content:
```

**Review by laymonage — COMMENTED** [review]
Generally looks good to me! Just one small comment

**Review by albinazs — COMMENTED** [review]
Looks good to me as well

File: docs/advanced_topics/accessibility_considerations.md (hunk 2 above)
Comment (thibaudcolas) [reply docs/advanced_topics/accessibility_considerations.md:135]: Good point :)

**Pull request merged**